Sources/WalletConnectRelay/RelayClient.swift

```diff
@@ -80,24 +80,14 @@ public final class RelayClient {
         logger: ConsoleLogging = ConsoleLogger(loggingLevel: .off)
     ) {
         let clientIdStorage = ClientIdStorage(keychain: keychainStorage)
-        let socketAuthenticator = SocketAuthenticator(
+        let dispatcher = Dispatcher(
+            relayHost: relayHost,
+            projectId: projectId,
             clientIdStorage: clientIdStorage,
```

Comment:
would adding RelayClientFactory, like we have for other clients solve the init problem for Dispatcher and RelayClient?